### ISSUE TITLE
One-class learning

### ISSUE THREAD

**Issue by basque21** [amaiya/ktrain#121]
Hi again, I'm trying to train a One-class scorer, but I'm not sure I'm understanding well how it works.

I trained the topic model with the whole dataset `tm = ktrain.text.get_topic_model(texts, n_features=10000)` and built it `tm.build(texts, threshold=0.0)`. But then I want to train the scorer with the documents with certain label, not based in the topics. I do `tm.train_scorer(doc_ids=tech_ids)`, where tech_ids are the ids of the texts with interested label. But then if I try to score, say, a sample of 1000 documents where 200 were used as seed in the `train_scorer`, these documents are not ranked with the highest values. How is it possible? I mean, if they are the seed, shouldn't they be  the most similar to them?

Thanks for the answer again... I'm a bit confused...

**EDIT**
Well, I have realized that when I'm feeding doc_ids, the scorer is getting all the topics those documents belong, am I correct?

So, in order to get my one-class classifier what do you think I should do? Train the get_topic_model with only the documents of this class and force to a unique topic?

**Comment by amaiya** [comment]
Hello @basque21 :   

The One-Class similarity scorer is implemented using scitkit-learn's LocalOutlierFactor.  Per the sklearn documentation, you should only make predictions on new, unseen data, not on examples used to train the scorer: From [sklearn documentation](https://scikit-learn.org/stable/modules/generated/sklearn.neighbors.LocalOutlierFactor.html):

> In this case be aware that that you should only use predict, decision_function and score_samples on new unseen data and not on the training set.

For these reasons, you'll notice that, in the example notebook, the scorer only ever scores documents **outside** of the seed set.

This may be why you're getting unexpected results when scoring documents in the seed set.
